app/client/src/sagas/ActionExecution/PromiseActionSaga.ts

```
@@ -79,9 +82,10 @@ export default function* executePromiseSaga(
         },
         responseData: catchArguments,
       });
-    } else {
-      throw new TriggerFailureError("Uncaught promise rejection", e);
     }
+    // This error could be a TriggerEvaluationError as well so
+    // throwing these errors as it is
+    throw e;
```

should we throw an error only when there's no `catch` block?

---

For e.g here we do have a catch(the showAlert call), but there is an error in then( assuming no query named Query1s exists). So in this case we were showing a toast but this error wasn't thrown back to `initiateActionTriggerExecution` saga. To show this error in the debugger the error needs to be accessible there so I am throwing it at the end of the saga if there are any errors.
```
{{Query1.run(() => Query1s.run(), () => showAlert("Test"))}}
```

---

Just checking: would it be displaying an error toast even when there's an explicit catch block added by the user?

---

Yes it is shown. This happens on release as well.